validator/remote/src/main/java/tech/pegasys/teku/validator/remote/typedef/handlers/SendSignedAttestationsRequest.java

```diff
@@ -13,29 +13,51 @@
 
 package tech.pegasys.teku.validator.remote.typedef.handlers;
 
+import static java.util.Collections.emptyMap;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.HEADER_CONSENSUS_VERSION;
 import static tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition.listOf;
+import static tech.pegasys.teku.validator.remote.apiclient.ValidatorApiMethod.SEND_SIGNED_ATTESTATION_V2;
 import static tech.pegasys.teku.validator.remote.typedef.FailureListResponse.getFailureListResponseResponseHandler;
 
 import java.util.Collections;
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
 import okhttp3.HttpUrl;
 import okhttp3.OkHttpClient;
 import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecMilestone;
 import tech.pegasys.teku.spec.datastructures.operations.Attestation;
 import tech.pegasys.teku.validator.api.SubmitDataError;
 import tech.pegasys.teku.validator.remote.apiclient.ValidatorApiMethod;
 import tech.pegasys.teku.validator.remote.typedef.FailureListResponse;
 
 public class SendSignedAttestationsRequest extends AbstractTypeDefRequest {
+
+  private final Spec spec;
+
   public SendSignedAttestationsRequest(
-      final HttpUrl baseEndpoint, final OkHttpClient okHttpClient) {
+      final HttpUrl baseEndpoint, final OkHttpClient okHttpClient, final Spec spec) {
     super(baseEndpoint, okHttpClient);
+    this.spec = spec;
   }
 
   public List<SubmitDataError> submit(final List<Attestation> attestations) {
     if (attestations.isEmpty()) {
       return Collections.emptyList();
     }
+    // Use attestation v2 api post Electra only. This logic can be removed once we reach the Electra
+    // milestone
+    final SpecMilestone specMilestone =
+        spec.atSlot(attestations.getFirst().getData().getSlot()).getMilestone();
+    if (specMilestone.isGreaterThanOrEqualTo(SpecMilestone.ELECTRA)) {
+      return submitPostElectra(attestations, specMilestone);
```

Comment:
here I remember the discussion we had about versioning data structures. My position was that we need to send always ELECTRA if data structure remains unchanged in next forks.

Reply:
Ok we clarified this